swiss_locator/core/filters/swiss_locator_filter.py

```diff
@@ -427,6 +431,79 @@ def triggerResult(self, result: QgsLocatorResult):
             if self.settings.value("show_map_tip"):
                 self.show_map_tip(swiss_result.layer, swiss_result.feature_id, point)
 
+        # Vector tiles
+        elif type(swiss_result) == VectorTilesLayerResult:
+            params = dict()
+            params["styleUrl"] = swiss_result.style or ""
+            params["url"] = swiss_result.url
+            params["type"] = "xyz"
+            params["zmax"] = "14"
+            params["zmin"] = "0"
+
+            url_with_params = "&".join([f"{k}={v}" for (k, v) in params.items()])
+
+            self.info(f"Loading layer: {url_with_params}")
+            vt_layer = QgsVectorTileLayer(url_with_params, result.displayString)
+
+            if not vt_layer.isValid():
+                msg = self.tr(
+                    "Cannot load Vector Tiles layer: {}".format(
+                        swiss_result.title
+                    )
+                )
+                level = Qgis.Warning
+                self.info(msg, level)
+            else:
+                vt_layer.setLabelsEnabled(True)
+                vt_layer.loadDefaultMetadata()
+
+                error, warnings = '', []
+                res, sublayers = vt_layer.loadDefaultStyleAndSubLayers(error, warnings)
+
+                if sublayers:
+                    msg = self.tr(
+                        "Sublayers found ({}): {}".format(
+                            swiss_result.title,
+                            "; ".join([sublayer.name() for sublayer in sublayers])
+                        )
+                    )
+                    level = Qgis.Info
+                    self.info(msg, level)
+                if error or warnings:
+                    msg = self.tr(
+                        "Error/warning found while loading default styles and sublayers for layer {}. Error: {} Warning: {}".format(
+                            swiss_result.title,
+                            error,
+                            "; ".join(warnings)
+                        )
+                    )
+                    level = Qgis.Warning
+                    self.info(msg, level)
+
+                msg = self.tr(
+                    "Layer added to the map: {}".format(
+                        swiss_result.title
+                    )
+                )
+                level = Qgis.Info
+                self.info(msg, level)
+
+                # Load basemap layers at the bottom of the layer tree
+                root = QgsProject.instance().layerTreeRoot()
+                if sublayers:
+                    # Sublayers should load on top of the vector tiles layer
+                    # We group them to keep them all together
+                    group = root.insertGroup(-1, vt_layer.name())
+                    for sublayer in sublayers:
+                        QgsProject.instance().addMapLayer(sublayer, False)
```

Comment:
I would suggest to use `addMapLayers` with all layers at once https://qgis.org/pyqgis/3.34/core/QgsProject.html#qgis.core.QgsProject.addMapLayers